pulsar-sql/presto-pulsar/src/main/java/org/apache/pulsar/sql/presto/PulsarRecordCursor.java

```diff
@@ -441,9 +442,11 @@ public boolean advanceNextPosition() {
         //start time for deseralizing record
         metricsTracker.start_RECORD_DESERIALIZE_TIME();
 
-        SchemaInfo schemaInfo;
+        SchemaInfo schemaInfo = getBytesSchemaInfo(pulsarSplit.getSchemaType(), pulsarSplit.getSchemaName());
```

Comment:
I noticed the current `getSchemaVersion` in the `RawMessageImpl` return null 

```
@Override
    public byte[] getSchemaVersion() {
        if (msgMetadata != null && msgMetadata.getMetadata().hasSchemaVersion()) {
            return msgMetadata.getMetadata().getSchemaVersion();
        } else {
            return null;
        }
    }
```

It should not throw `java.nio.BufferUnderflowException`. Could you please confirm here?

Reply:
This method may cause the exception `java.nio.BufferUnderflowException`.

```
    private SchemaInfo loadSchema(BytesSchemaVersion bytesSchemaVersion) throws PulsarAdminException {
        ClassLoader originalContextLoader = Thread.currentThread().getContextClassLoader();
        try {
            Thread.currentThread().setContextClassLoader(InjectionManagerFactory.class.getClassLoader());
            return pulsarAdmin.schemas()
                    .getSchemaInfo(topicName.toString(), ByteBuffer.wrap(bytesSchemaVersion.get()).getLong());
        } finally {
            Thread.currentThread().setContextClassLoader(originalContextLoader);
        }
    }

```

The schema version of `Schema.BYTES` is an empty bytes array. So `ByteBuffer.wrap(bytesSchemaVersion.get()).getLong()` will cause the exception `java.nio.BufferUnderflowException`.

Reply:
I see.